src/react-utils/useSticky.ts

```diff
@@ -1,45 +1,139 @@
-import {RefObject, useState} from 'react';
-
-import {useEffectOnce, useLatest} from 'react-use';
-
-const events: ReadonlySet<keyof WindowEventMap> = new Set<keyof WindowEventMap>([
-    'resize',
-    'scroll',
-    'touchstart',
-    'touchmove',
-    'touchend',
-    'pageshow',
-    'load',
-    'orientationchange',
-]);
-
-export function useSticky<T extends HTMLElement>(elemRef: RefObject<T>) {
+import {RefObject, useEffect, useState} from 'react';
+
+import throttle from 'lodash/throttle';
+
+/**
+ * Finds the closest parent element with overflow: auto or overflow: scroll.
+ *
+ * This function traverses up the DOM tree starting from the given element,
+ * checking each parent element for overflow: auto or overflow: scroll styles.
+ * If such an element is found, it is returned as the scroll container.
+ * If no matching element is found, the global window object is returned.
+ *
+ * @param {HTMLElement | null} element - The starting element from which to begin the search.
+ * @returns {HTMLElement | Window} - The first parent with overflow: auto/scroll, or window.
+ */
+const findScrollContainer = (element: HTMLElement | null): HTMLElement | Window => {
+    let currentElement = element;
+    while (currentElement) {
+        const overflow = window.getComputedStyle(currentElement).overflow;
+        if (overflow === 'auto' || overflow === 'scroll') {
+            return currentElement;
+        }
+        currentElement = currentElement.parentElement;
+    }
+    return window;
+};
+
+/**
+ * Finds the value of a CSS variable starting from the specified element.
+ * If not found locally, it will fallback to the global :root.
+ *
+ * @param {HTMLElement | null} element - The starting element to search for the CSS variable.
+ * @param {string} variableName - The name of the CSS variable to search for.
+ * @returns {number} - The value of the CSS variable or 0 if not found.
+ */
+const findCssVariableValue = (element: HTMLElement | null, variableName: string): number => {
+    let currentElement = element;
+    while (currentElement) {
+        const value = getComputedStyle(currentElement).getPropertyValue(variableName);
+        if (value) {
+            return parseFloat(value);
```

Comment:
If you need to calculate numeric value of css-var, you need to use something like [this function](https://github.com/gravity-ui/markdown-editor/blob/bd85a1c4c073c665c7fa9b8c7277a8441c6f378b/src/bundle/Editor.ts#L538) because is css-vars can store a css expression like `5px + 7px`, so parseFloat or parseInt don't work correctly